libs/vkd3d/swapchain.c

```diff
@@ -1626,6 +1730,10 @@ static bool dxgi_vk_swap_chain_request_needs_swapchain_recreation(
     return request->dxgi_color_space_type != last_request->dxgi_color_space_type ||
             request->dxgi_format != last_request->dxgi_format ||
             request->target_min_image_count != last_request->target_min_image_count ||
+            request->low_latency_update_requested ||
+            (chain->present.low_latency_state.mode &&
+                    (request->frame_id != chain->present.previous_application_frame_id) &&
+                    !chain->present.using_application_frame_id) ||
```

Comment:
I think we should recreate the swapchain also when switching from app marker IDs to total count of Present() calls, which happens when turning off Reflex. The Reflex documentation only says: "The frameID can start at an abitrary moment in the application lifetime but must strictly increment", and we don't know if it's larger or smaller than present count.

With this considered, I think we should 1. switch to Present() count too when we detect app ID is not incrementing, and 2. recreate swapchain whenever we switch between Present() count and app marker ID. That is, with the following helper:
```c
bool dxgi_vk_swap_chain_can_use_app_frame_id(struct dxgi_vk_swap_chain *chain, struct dxgi_vk_swap_chain_present_request *request) {
    return chain->present.low_latency_state.mode &&
      request->frame_id > chain->present.previous_application_frame_id;
}
```
and recreate whenever the return value changes.

Reply:
I think this should be resolved with the most recent update. The code uses the present tasks present_count now for the present id when the application isn't providing a frame id.